indico/modules/events/papers/notifications.py

```diff
@@ -0,0 +1,99 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import session
+
+from indico.core.notifications import make_email, send_email
+from indico.modules.events.papers.settings import paper_reviewing_settings, PaperReviewingRole
+from indico.web.flask.templating import get_template_module
+
+
+def notify_paper_revision_submission(revision):
+    event = revision.paper.event_new
+    roles_to_notify = paper_reviewing_settings.get(event, 'notify_on_paper_submission')
+    if PaperReviewingRole.judge in roles_to_notify:
+        for judge in revision.paper.contribution.paper_judges:
+            template = get_template_module('events/papers/emails/revision_submission_to_judge.html', event=event,
+                                           revision=revision, receiver=judge)
+            email = make_email(to_list=judge.email, template=template, html=True)
+            send_email(email, event=event, module='Papers', user=session.user)
+    reviewers = set()
+    if PaperReviewingRole.layout_reviewer in roles_to_notify:
+        reviewers = revision.paper.cfp.layout_reviewers
```

Comment:
I'd use `|=` for consistency. That way you also guarantee  that the next `|=` won't change anything (imagine if the `layout_reviewers` property returned something that's not generated on the fly - then the in-place change 2 lines below could end up modifying data in the DB)